src/main/scala/diplomacy/Nodes.scala

```diff
@@ -545,15 +595,27 @@ class MixedAdapterNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
   override def description = "adapter"
   protected[diplomacy] override def flexibleArityDirection = true
   protected[diplomacy] def resolveStar(iKnown: Int, oKnown: Int, iStars: Int, oStars: Int): (Int, Int) = {
-    require (oStars + iStars <= 1, s"$context appears left of a :*= $iStars times and right of a :=* $oStars times; at most once is allowed")
+    require (oStars + iStars <= 1,
+      s"""$context
+         |$bindingInfo
+         |appears left of a :*= $iStars times and right of a :=* $oStars times; at most once is allowed""".stripMargin)
```

Comment:
`| The following node appears inwards of a :*= $iStars times and outwards of a :=* $oStars times; at most one of either is allowed."`

Reply:
I think left/right here should be more intuitional?

Reply:
Left/right might be more clear in terms of what to look for in the source code. Inwards/Outwards is more clear in terms of what diplomacy calls things, like `DI` `UO` etc. But maybe for an error message left/right is better? 

But what if I have:
```
a :*= b
b  :*=* c
c :*=* d
d :=* e
```
`c` will get the error, but is c "right" of `:*=` or "left" of `:=*`. Maybe "transitively left"?

Reply:
I see you have lots of `lefts` and `rights` already, I don't think you have to change them again. Let's try this out as it is now.